src/analysis/normed_space/operator_norm.lean

```diff
@@ -41,115 +42,100 @@ instance bounded_linear_maps.to_fun : has_coe_to_fun $ L(E,F) :=
-let ⟨c, _, _⟩ := (exists_bound A : ∃ c, c > 0 ∧ ∀ x : E, ∥ A x ∥ ≤ c * ∥ x ∥) in
-have ∥A x∥ ∈ (image (norm ∘ A) {x | ∥x∥ ≤ 1}), from mem_image_of_mem _ $ le_of_eq ‹∥x∥ = 1›,
-le_cSup (norm_of_unit_ball_bdd_above A) ‹∥A x∥ ∈ _›
+begin
+  have : (0 : ℝ) = (λ x, ∥A x∥/∥x∥) 0, by simp,
+  rw this,
+  exact le_csupr (bdd_above_range_norm_image_div_norm A),
+end
 
 /-- This is the fundamental property of the operator norm: ∥A x∥ ≤ ∥A∥ * ∥x∥. -/
 theorem bounded_by_operator_norm {A : L(E,F)} {x : E} : ∥A x∥ ≤ ∥A∥ * ∥x∥ :=
-have A 0 = 0, from (to_linear_map A).map_zero,
-classical.by_cases
-  (assume : x = (0:E),
-    show ∥A x∥ ≤ ∥A∥ * ∥x∥, by rw[‹x = 0›, ‹A 0 = 0›, norm_zero, norm_zero, mul_zero]; exact le_refl 0)
-  (assume : x ≠ (0:E),
-    have ∥x∥ ≠ 0, from ne_of_gt $ (norm_pos_iff x).mpr ‹x ≠ 0›,
-    have ∥∥x∥⁻¹∥ = ∥x∥⁻¹, from abs_of_nonneg $ inv_nonneg.mpr $ norm_nonneg x,
-    have ∥∥x∥⁻¹•x∥ = 1, begin rw[norm_smul, ‹∥∥x∥⁻¹∥ = ∥x∥⁻¹›], exact inv_mul_cancel ‹∥x∥ ≠ 0› end,
-    calc ∥A x∥ = (∥x∥ * ∥x∥⁻¹) * ∥A x∥ : by rw[mul_inv_cancel ‹∥x∥ ≠ 0›]; ring
-          ... = ∥∥x∥⁻¹∥ * ∥A x∥ * ∥x∥  : by rw[‹∥∥x∥⁻¹∥ = ∥x∥⁻¹›]; ring
-          ... = ∥∥x∥⁻¹• A x ∥ * ∥x∥    : by rw[←normed_space.norm_smul ∥x∥⁻¹ (A x)]
-          ... = ∥A (∥x∥⁻¹• x)∥ * ∥x∥   : by {
-                                          change  ∥∥x∥⁻¹ • A.val x∥ * ∥x∥ = ∥A.val (∥x∥⁻¹ • x)∥ * ∥x∥,
-                                          rw A.property.smul}
-          ... ≤ ∥A∥ * ∥x∥              : (mul_le_mul_right ((norm_pos_iff x).mpr ‹x ≠ 0›)).mpr
-                                          (bounded_by_operator_norm_on_unit_vector A ‹∥∥x∥⁻¹•x∥ = 1›))
-
-lemma bounded_by_operator_norm_on_unit_ball (A : L(E, F)) {x : E} (_ : ∥x∥ ≤ 1) : ∥A x∥ ≤ ∥A∥ :=
-calc ∥A x∥ ≤ ∥A∥ * ∥x∥ : bounded_by_operator_norm
-        ... ≤ ∥A∥ * 1 : mul_le_mul_of_nonneg_left ‹∥x∥ ≤ 1› (operator_norm_nonneg A)
-        ... = ∥A∥ : mul_one ∥A∥
-
-lemma operator_norm_bounded_by {A : L(E,F)} (c : nnreal) :
-  (∀ x : E, ∥x∥ ≤ 1 → ∥A x∥ ≤ (c:ℝ)) → ∥A∥ ≤ c :=
-assume H : ∀ x : E, ∥x∥ ≤ 1 → ∥A x∥ ≤ c,
-suffices Sup (image (norm ∘ A) {x | ∥x∥ ≤ 1}) ≤ c, by assumption,
-cSup_le (set.ne_empty_of_mem $ zero_in_im_ball A)
-  (show ∀ (r : ℝ), r ∈ (image (norm ∘ A) {x | ∥x∥ ≤ 1}) → r ≤ c, from
-    assume r ⟨x, _, _⟩,
-      calc r = ∥A x∥ : eq.symm ‹_›
-         ... ≤ c : H x ‹_›)
-
-theorem operator_norm_triangle (A : L(E,F)) (B : L(E,F)) : ∥A + B∥ ≤ ∥A∥ + ∥B∥ :=
-operator_norm_bounded_by (⟨∥A∥, operator_norm_nonneg A⟩ + ⟨∥B∥, operator_norm_nonneg B⟩)
-  (assume x _, by exact
+begin
+  classical, by_cases h : x = 0,
+  { simp [h] },
+  { have : ∥A x∥/∥x∥ ≤ ∥A∥, from le_csupr (bdd_above_range_norm_image_div_norm A),
+    rwa (div_le_iff _) at this,
+    have : ∥x∥ ≠ 0, from ne_of_gt ((norm_pos_iff x).mpr h),
+    have I := le_iff_eq_or_lt.1 (norm_nonneg x),
+    simpa [ne.symm this] using I }
+end
+
+/-- Conversely, if one controls the norm of every A x, then one controls the norm of A. -/
```

Comment:
This docstring probably doesn't make much sense as a tooltip, because then the "Conversely" is a bit out of context.

Reply:
Indeed. Updated